codecov/__init__.py

```diff
@@ -719,7 +719,7 @@ def main(*argv, **kwargs):
                         result, upload_url = res[0], res[1]
 
                         write('    Uploading to S3...')
-                        s3 = requests.put(upload_url, data=reports,
+                        s3 = requests.put(upload_url, data=reports.encode('utf-8'),
```

Comment:
This is bad because `reports` is `bytes` in python2 and `string` in python3. Only `string` must be encoded to bytes.

Alternatively would be to make `reports` a `string` also in python2 (there it is called `unicode`). You can use `u''` prefix for that.